### PR TITLE
fix: apply config policy constraints to PATCH /experiments/:id

### PR DESCRIPTION
**Pull request by stoksc**
## Ticket
CM-564

## Description
Pushing this check down a level makes sure it applies for `PATCH /api/v1/experiments/:id` calls, too.

## Test Plan
TBD.

### CODE REVIEW

**Review by kkunapuli — APPROVED** [review]
Looks great - thanks for finding and fixing this!

**Review by ShreyaLnuHpe — APPROVED** [review]
LGTM!

File: master/internal/experiment.go [hunk 1]
```diff
@@ -1033,6 +1034,29 @@ func (e *internalExperiment) setPriority(priority *int, forward bool) (err error
...
+
+	workspaceModel, err := workspace.WorkspaceByProjectID(context.TODO(), e.ProjectID)
+	if err != nil && errors.Cause(err) != sql.ErrNoRows {
+		return err
+	}
+	wkspID := resolveWorkspaceID(workspaceModel)
+
+	// Returns an error if RM does not implement priority.
+	if smallerHigher, err := e.rm.SmallerValueIsHigherPriority(); err == nil {
+		ok, err := configpolicy.PriorityAllowed(
```
Comment (amandavialva01) [inline master/internal/experiment.go:1046]: This looks great! Same comment as [here](https://github.com/determined-ai/determined/pull/10047#discussion_r1799456338)

**Review by amandavialva01 — COMMENTED** [review]
This looks great! Same comment as [here](https://github.com/determined-ai/determined/pull/10047#discussion_r1799456338) 
Also, can we change the PR title to "apply priority constraints and check invariant config in PATCH /experiments/:id" (or something like that) after suggested changes are made ?

**Pull request merged**